### PR TITLE
Refresh

### PR DESCRIPTION
**Pull request by ddgenome**
Opening this pull request. We'll be merging this tomorrow, so let us know what you think.

Note also #202 and #205

### CODE REVIEW

**Comment by ddgenome** [comment]
Thanks @kipz ! I'm going to start addressing these many, many comments.

File: docs/index.md [hunk 1]
```diff
@@ -1,148 +1,125 @@
...
+concepts describe the pieces Atomist provides to make it easy for you
+to create automations that reduce drudgery and distractions.
+
+### Development automation platform
+
+Atomist's development automation platform is powered by a service that
+ingests and correlates events from your software development flow.  At
+the heart of the service is a single coherent model of your
+development flow: code, people, and processes.  You query and mutate
+that model using [GraphQL][gql] when you write your own automations.
```
Comment (whostolebenfrog) [inline docs/index.md]: Repetition of "flow" here: I'd go with

"At the heart of the service is a single coherent model ~~of your development flow~~: code, people, and processes.

File: docs/index.md [hunk 2]
```diff
@@ -1,148 +1,125 @@
...
+
+[gql]: http://graphql.org/ (GraphQL)
+
+### Automation client
+
+Code in automations is written in Atomist automation clients, which
```
Comment (whostolebenfrog) [inline docs/index.md]: Link https://github.com/atomist/automation-client-ts ?

File: docs/index.md [hunk 3]
```diff
@@ -1,148 +1,125 @@
...
+interact with the Atomist platform using primarily GraphQL.  Each
+client is itself a server which hosts automations that can be invoked
```
Comment (whostolebenfrog) [inline docs/index.md]: I wonder if the client server terminology here is a bit confusing.

Could remove the server part. "Each client hosts automations that can be invoked"

File: docs/index.md [hunk 4]
```diff
@@ -1,148 +1,125 @@
...
+via the Atomist bot, Slack buttons, or when events occur.  A client
+can host any number of automations, and can be hosted wherever the
+author likes: locally during testing, inside a corporate firewall, or
+on a public cloud or PaaS.
+
+### Commands
+
+Commands perform actions when invoked by users by sending the Atomist
```
Comment (whostolebenfrog) [inline docs/index.md]: Commands perform actions when invoked by users ~~by~~ sending the Atomist
bot messages, click **ing** buttons in Slack, or using the Atomist CLI.

File: docs/index.md [hunk 5]
```diff
@@ -1,148 +1,125 @@
...
+bot messages, click buttons in Slack, or using the Atomist CLI.
+Command implementations are defined
+in [**command handlers**][command].
+
+Examples of commands include the bot command `create issue` and the
```
Comment (whostolebenfrog) [inline docs/index.md]: +Examples of commands include the bot command `create issue` and the
 +`merge pull request` command that users invoke by pressing a button in
 +Slack.

Examples of commands include the bot commands `create issue` or `merge pull request` where a user can invoke the command by clicking a button in slack.

File: docs/index.md [hunk 6]
```diff
@@ -1,148 +1,125 @@
...
+`merge pull request` command that users invoke by pressing a button in
+Slack.
+
+[command]: automations/commands.md (Bot Commands)
+
+### Events
+
+Events can be anything from commits to pushes to CI builds to
```
Comment (whostolebenfrog) [inline docs/index.md]: > +Events can be anything from commits to pushes to CI builds to
 +deployments to stack traces in production logs.  When an event is
 +ingested, Atomist relates it to other events to build up necessary
 +contextual information.  Atomist may also take automated action. The
 +response to an event is defined in
 +an [**event handler**][event].

I think this paragraph is a bit confusing. Particularly "Atomist may also take automated action" - I think we mean that the users atomist system can take automated actions based on these events.

Maybe rephrase like this:

Events can be sourced from anywhere: from commits, pushes or CI builds to deployments or stack traces from production logs. When an event is ingested, Atomist relates it to other events to build up contextual information. A push contains commits, it triggered a build in your CI system eventually leading to a deployment. Atomist is able to take automated actions based on these events. The response to an event is defined in an [**event handler**][event].

File: docs/index.md [hunk 7]
```diff
@@ -1,148 +1,125 @@
...
+deployments to stack traces in production logs.  When an event is
+ingested, Atomist relates it to other events to build up necessary
+contextual information.  Atomist may also take automated action. The
+response to an event is defined in
+an [**event handler**][event].
+
+How does a handler know what event it should act on?  GraphQL
+subscriptions are used to match trigger criteria.  For example, you
+can create a subscription to receive an event when Kubernetes
+deployment results in a pod crash looping and connect that to an event
```
Comment (whostolebenfrog) [inline docs/index.md]: > deployment results

to

deployments result

File: docs/index.md [hunk 8]
```diff
@@ -1,148 +1,125 @@
...
+
+Events can be anything from commits to pushes to CI builds to
+deployments to stack traces in production logs.  When an event is
+ingested, Atomist relates it to other events to build up necessary
+contextual information.  Atomist may also take automated action. The
+response to an event is defined in
+an [**event handler**][event].
+
+How does a handler know what event it should act on?  GraphQL
+subscriptions are used to match trigger criteria.  For example, you
```
Comment (whostolebenfrog) [inline docs/index.md]: Can we link something here? Right now we say "GraphQL subscriptions are used to match trigger criteria" but we don't explain it. I think we either need to link, add a paragraph or hold off introducing this concept here.

File: docs/index.md [hunk 9]
```diff
@@ -1,148 +1,125 @@
...
+### Integrations
+
+Atomist receives events from and performs operations on many systems,
+including version control, CI systems, Slack, and more.
+
+Atomist uses the native integration technology for each platform or
+tool.  For example, to integrate with GitHub and Travis CI, Atomist
+uses webhooks; to integrate with Slack it uses their native real-time
+messaging (RTM) API.  For each platform Atomist integrates with, it
+asks for the minimal set of permissions required to perform its
```
Comment (whostolebenfrog) [inline docs/index.md]: drop "its duties"

File: docs/automations/commands.md [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+-   a name for the command handler
+
+In this guide you'll create MyCommandHandler, which responds to "do my thing".
+
+## Command handler
+
+Command handlers are classes with a `handle` method
+ and some decorators that supply metadata.
+Store them anywhere in the `src` directory;
+your automation client discovers them on startup. (Or [specify them yourself](client.md#client-configuration).)
```
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/automations/commands.md]: The first close parenthesis is for the markdown link, the second actually closes the parenthetical clause.

[the remark this answers is not in the document]
Comment (kipz) [reply docs/automations/commands.md]: I get that. I was referring to starting a new sentence with parenthsese, which I thought wasn't proper.

**Comment by whostolebenfrog** [comment]
As a general thought - the index.md page leads people through a high level of our concepts. But at the end it doesn't naturally lead people onto the next page. Where should people go from here? What's the next page to read? Could we link to the most appropriate getting started doc from the bottom?

File: docs/automations/graphql.md [hunk 11]
```diff
@@ -0,0 +1,227 @@
+[GraphQL](http://graphql.org) is a powerful query language that you use to query
+and mutate your data in the Atomist automation platform.
+
+Besides being a great query language, GraphQL provides great tool support based
+on strongly-typed schemas, type generation for TypeScript, and many other
```
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/automations/graphql.md]: both are acceptible

File: docs/automations/graphql.md [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+queries and displays the shape of the resulting data. It also provides access to the data model's documentation.
+
+!!! note
+    On Mac OS it's easiest to install GraphiQL as a stand-alone app
+    from https://github.com/skevy/graphiql-app.
+
+After installation you need to configure the endpoint and authentication in GraphiQL.
+
+To setup authentication, click on _Edit HTTP Headers_ and select _+ Add Header_.
+Enter `Authorization` as _Header name_ and `Bearer <token>` as _Header value_.
```
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/automations/graphql.md]: @cdupuis ?

File: docs/automations/index.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
+Atomist ingests events from your software development systems
+and makes them available via an automation API:
```
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/automations/index.md]: We haven't anywhere else.

File: docs/automations/index.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+
+-   Source code repositories like [GitHub.com][gh]
+    and [GitHub Enterprise][ghe]
+-   Issue tracking systems like [GitHub][gh-issues] and [Jira][jira]
+-   Continuous integration platforms
```
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/automations/index.md]: I don't think it is a proper noun. "Continuous" is capitalized because we use sentence case for lists.

File: docs/automations/index.md [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+Automations rely on this data model to provide the necessary context so they can always
+do the right thing.
+
+The Atomist development automation platform also provides a simple yet
+powerful interface for implementing custom bot commands.  By providing
+all the infrastructure needed to hear commands, collect parameters,
+execute the code, and respond back, Atomist lets you focus on writing
+your bot command code, not all the ceremony around it.
+
+The Atomist automation API can be accessed via any compliant client.
```
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/automations/index.md]: At present we are not thinking of it as a proper noun.

File: docs/automations/paas.md [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+
+Running the automation client locally is extremely helpful
+during development of your automations.
+
+-   You can debug commands and event handlers using local development
+    tools like Visual Studio Code and Google Chrome.
+-   You can iterate rapidly because there is no deployment and only a
+    minimal build process.
+
+If you bootstrapped your client project using an Atomist seed and
```
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/automations/paas.md]: Good to keep in mind for when we actually have one.

File: docs/automations/prerequisites.md [hunk 17]
```diff
@@ -0,0 +1,115 @@
+Before you begin developing and running your own automations, you
+need to have Atomist [set up in your team][setup] and know your Slack
+team ID, which the Atomist Bot is happy to tell you.  Just send the
+`team` message to the Atomist Bot (be sure the bot is already in the
+channel!).
```
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/automations/prerequisites.md]: "Atomist bot" it is!

File: docs/automations/prerequisites.md [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+
+The reference implementation of the Atomist automation API client is
+implemented in [TypeScript][ts], a superset of [JavaScript][js].  To develop and
+run automations using the reference implementation of the automation
+client, you must install Node.js.  The easiest way to install
+Node.js is to go to the [Node.js web site][node] and follow the
+installation instructions for your platform.  This makes the
+`node` and `npm` programs available on your system.
+
+Alternatively, macOS users with [Homebrew][brew]
```
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/automations/prerequisites.md]: fixed the other to be "macOS"

File: docs/automations/slack.md [hunk 19]
```diff
@@ -0,0 +1,314 @@
...
+    confirm: {
+        title: "Search Again?",
+        text: "Do you really want to run the search again?",
+        dismiss_text: "No",
+        ok_text: "Yes"
+    },
+};
+```
+
+With the following, you're preparing a [command handler](commands.md)  and its parameter to be
```
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/automations/slack.md]: HTML will

File: docs/support.md [hunk 20]
```diff
@@ -0,0 +1,25 @@
+Thanks so much for choosing Atomist!  At Atomist, want all
```
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/support.md]: I had "Thank you", @WillGardella changed it to "Thanks" preferring a more informal tone.

File: docs/index.md (hunk 9 above)
Comment (ddgenome) [reply docs/index.md]: Just "required to perform"? That seems abrupt.

Comment (whostolebenfrog) [reply docs/index.md]: Maybe:

"For each platform Atomist integrates with it requests smallest set of permissions that are required to work"

Or drop the whole sentence? It's actually quite a high bar and easy to miss.

Also in addition to the 3 other comments above I notice that it's marked 5 more as outdated that aren't outdated - not sure why!

File: docs/index.md (hunk 8 above)
Comment (ddgenome) [reply docs/index.md]: It is linked a few paragraphs above.

File: docs/index.md (hunk 9 above)
Comment (ddgenome) [reply docs/index.md]: You made them while I was fixing @kipz comments on/around the same lines. I noticed this too, expanded the comments, and addressed them all, I think. This one, I'm inclined to leave for @WillGardella to do something about.

File: docs/automations/commands.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (ddgenome) [reply docs/automations/commands.md]: I think @WillGardella is going through this doc now.

**Comment by ddgenome** [comment]
I addressed this request @whostolebenfrog and your review comments.
